search-service/src/main/kotlin/com/egm/stellio/search/service/EntityAccessRightsService.kt

```diff
@@ -175,6 +179,35 @@ class EntityAccessRightsService(
         }
     }
 
+    suspend fun hasReadRightOnEntity(
+        sub: Option<Sub>,
+        entityId: URI,
+        specificAccessPolicies: List<AuthContextModel.SpecificAccessPolicy>
+    ): Either<APIException, Unit> {
+        val hasRight = temporalEntityAttributeService.hasSpecificAccessPolicies(
+            entityId,
+            specificAccessPolicies
+        ).awaitFirst() || canReadEntity(sub, entityId).awaitFirst()
```

Comment:
you could call directly `checkHasAccessRight` since the `canRead` / `canWrite` just do this (so they are a useless intermediate now)

Reply:
also this is missing a bit of consistency: you have one security check here and all the others in `checkHasAccessRight`, wouldn't be better to have all security checks there?